src/newtype_macros.rs

```diff
@@ -75,12 +75,27 @@ macro_rules! newtype_traits (($newtype:ident, $len:expr) => (
         }
     }
 
+    impl AsRef<[u8]> for $newtype {
+        #[inline]
+        fn as_ref(&self) -> &[u8] {
+            &self.0
+        }
+    }
+
+    impl AsMut<[u8]> for $newtype {
+        #[inline]
+        fn as_mut(&mut self) -> &mut [u8] {
```

Comment:
Does this impl provides a write access to a type that was previously impossible to do? Can I change the inner data of a newtype without this PR?

Reply:
You could do it before, using the index-implementation, or you could create a new one using `from_slice()`. I'm ok with that, given that people might want to read keys from files or other sources.

Reply:
You have to use `std::ops::IndexMut` to write into the struct. Was there such an impl before your PR?

Reply:
No, you’re right. There wasn’t.

Reply:
I think there should not be AsMut impl for newtype. If you want to change the struct, use ::from_slice and replace the whole struct.